extensions/wikia/AdEngine/js/lookup/rubiconFastlane.js

```diff
@@ -0,0 +1,221 @@
+/*global define*/
+define('ext.wikia.adEngine.lookup.rubiconFastlane', [
+	'ext.wikia.adEngine.adContext',
+	'ext.wikia.adEngine.adTracker',
+	'ext.wikia.adEngine.utils.adLogicZoneParams',
+	'wikia.document',
+	'wikia.log',
+	'wikia.window'
+], function (adContext, adTracker, adLogicZoneParams, doc, log, win) {
+	'use strict';
+
+	var called = false,
+		config = {
+			oasis: {
+				TOP_LEADERBOARD: {
+					sizes: [[728, 90], [970, 250]],
+					targeting: { loc: 'top' }
+				},
+				TOP_RIGHT_BOXAD: {
+					sizes: [[300, 250], [300, 600]],
+					targeting: { loc: 'top' }
+				},
+				LEFT_SKYSCRAPER_2: {
+					sizes: [[160, 600]],
+					targeting: { loc: 'middle' }
+				},
+				LEFT_SKYSCRAPER_3: {
+					sizes: [[160, 600]],
+					targeting: { loc: 'footer' }
+				},
+				INCONTENT_BOXAD_1: {
+					sizes: [[300, 250]],
+					targeting: { loc: 'middle' }
+				},
+				PREFOOTER_LEFT_BOXAD: {
+					sizes: [[300, 250]],
+					targeting: { loc: 'footer' }
+				},
+				PREFOOTER_RIGHT_BOXAD: {
+					sizes: [[300, 250]],
+					targeting: { loc: 'footer' }
+				}
+			},
+			mercury: {
+				MOBILE_IN_CONTENT: {
+					sizes: [[300, 250]]
+				},
+				MOBILE_PREFOOTER: {
+					sizes: [[300, 250]]
+				},
+				MOBILE_TOP_LEADERBOARD: {
+					sizes: [[320, 50]]
+				}
+			}
+		},
+		context = adContext.getContext(),
+		logGroup = 'ext.wikia.adEngine.lookup.rubiconFastlane',
+		priceMap = {},
+		response = false,
+		rubiconSlots = [],
+		slots = {},
+		timing;
+
+	function getSlots(skin) {
+		var pageType = context.targeting.pageType,
+			slotName;
+
+		slots = config[skin];
+		if (skin === 'oasis' && pageType === 'home') {
+			for (slotName in slots) {
+				if (slots.hasOwnProperty(slotName) && slotName.indexOf('TOP') > -1) {
```

Comment:
Maybe at least lets describe more what is happening here by moving this if to separate function, for example getHomePageSlots ?